src/core/shape_inference/include/tile_shape_inference.hpp

```diff
@@ -51,9 +52,10 @@ std::vector<T> shape_infer(const Tile* op,
                        rep_it,
                        std::back_inserter(output_shape),
                        std::multiplies<TDim>());
-    } else if (arg_rank.is_static() && repeats_shape[0].is_static()) {
-        // unknown repeats but shape is 1-D static, any dim can be repeated (add missing dimension)
-        output_shape.resize(std::max<size_t>(arg_rank.get_length(), repeats_shape[0].get_length()));
+    } else if (arg_rank.is_static()) {
+        // unknown repeats any dim can be repeated (add missing dimension)
+        const auto r = (repeats_rank.is_static() && repeats_shape[0].is_static()) ? repeats_shape[0].get_length() : 0;
+        output_shape.resize(std::max<size_t>(arg_rank.get_length(), r));
```

Comment:
This is incorrect behavior. Please see the specification. repeats could be longer than data rank and it will extend output rank.

Reply:
Changed to return dynamic shape.